engines/efh/efh.cpp

```diff
@@ -0,0 +1,2665 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "common/system.h"
+#include "common/random.h"
+#include "common/error.h"
+#include "common/events.h"
+#include "engines/util.h"
+
+#include "efh/efh.h"
+#include "efh/constants.h"
+
+namespace Efh {
+
+void EfhGraphicsStruct::copy(EfhGraphicsStruct *src) {
+	// Same buffer address
+	_vgaLineBuffer = src->_vgaLineBuffer;
+	_shiftValue = src->_shiftValue;
+	_width = src->_width;
+	_height = src->_height;
+	_area = src->_area;
+}
+
+bool InvObject::isEquipped() {
+	return (_stat1 & 0x80) != 0;
+}
+
+int8 InvObject::getUsesLeft() {
+	return _stat1 & 0x7F;
+}
+
+EfhEngine::~EfhEngine() {
+	delete _rnd;
+	delete _graphicsStruct;
+	delete _vgaGraphicsStruct1;
+	delete _vgaGraphicsStruct2;
+}
+
+Common::Error EfhEngine::run() {
+	debug("run");
+
+	initialize();
+	initGraphics(320, 200);
+
+	_mainSurface = new Graphics::Surface();
+	_mainSurface->create(320, 200, Graphics::PixelFormat::createFormatCLUT8());
```

Comment:
Are you ever calling the Surface::free method and then deleting the surface in shutdown?

Reply:
Right... Fixed now, thanks